packages/markers/src/browser/problem/problem-filter.tsx

```diff
@@ -0,0 +1,139 @@
+    const defaultValue = {
+        text: '',
+        showErrors: true,
+        showWarnings: true,
+        showInfos: true,
+        showHints: true,
+        activeFile: false,
+        useFilesExclude: false
+    };
+
+    const [filters, setFilters] = React.useState<Filters>({ ...defaultValue });
+    const [controlVaild, setControlVaild] = React.useState<boolean>(false);
+    const [markerStat, setMarkerStat] = React.useState({
+        total: 0,
+        filtered: 0
+    });
+
+    const changeHandler = (e: React.ChangeEvent<HTMLInputElement>, value: string) => {
+        const newFilter = {
+            ...filters,
+            text: value?.trim() ?? ''
+        };
+        setFilters(newFilter);
+    };
+
+    const clickHandler = (key: FilterKeys) => {
+        const newFilter = {
+            ...filters,
+            [key]: !filters[key]
+        };
+
+        setFilters(newFilter);
+    };
+
+    const isControlDefaultValue = () => Object.keys(defaultValue).filter(key => key !== 'text').every((key: FilterKeys) => filters[key] === defaultValue[key]);
+
+    const udpateProblemStat = () => {
+        const { getProblemStat } = props;
+        let total = 0;
+        let filtered = 0;
+        if (!isControlDefaultValue() || filters.text) {
+            total = getProblemStat(false);
+            filtered = getProblemStat(true);
+        }
+        setMarkerStat({
+            total,
+            filtered
+        });
+    };
+
+    React.useEffect(() => {
+        props.onChange?.(filters);
+        setControlVaild(!isControlDefaultValue());
+        udpateProblemStat();
+    }, [filters]);
+
+    React.useEffect(() => {
+        const { onDidChangeMarkersEvent } = props;
+        const listener = onDidChangeMarkersEvent(() => {
+            udpateProblemStat();
+        });
+        return () => listener.dispose();
+    }, [filters]);
+
+    const creatLabel = (label: string, isToggled = true) => <>
+        <span className={`theia-dropdown-component-dropdown-item-icon ${isToggled ? 'toggled' : ''}`}></span>
+        <span>{label}</span>
+    </>;
+
+    const options = React.useMemo(() => {
+        const opts = [
+            { key: 'showErrors', label: 'Show Errors', group: '0' },
+            { key: 'showWarnings', label: 'Show Warnings', group: '0' },
+            { key: 'showInfos', label: 'Show Infos', group: '0' },
+            { key: 'showHints', label: 'Show Hints', group: '0' },
+            { key: 'activeFile', label: 'Show Active File Only', group: '1' },
+            { key: 'useFilesExclude', label: 'Hide Excluded Files', group: '1' }
+        ];
+
+        return opts.map(({ key, label, group }) => ({
+            value: key,
+            label: creatLabel(key === 'showHints' ? nls.localize('theia/markers/showHints', label) : nls.localizeByDefault(label), filters[key as FilterKeys]),
+            group,
+            onClick: () => clickHandler(key as FilterKeys)
+        }));
+    }, [filters]);
```

Comment:
Suggestion: This makes it a bit hard for the static analysis to confirm the existance of the localization keys. I would suggest the following:

```suggestion
    const options = React.useMemo(() => {
        const opts = [
            { key: 'showErrors', label: nls.localizeByDefault('Show Errors'), group: '0' },
            { key: 'showWarnings', label: nls.localizeByDefault('Show Warnings'), group: '0' },
            { key: 'showInfos', label: nls.localizeByDefault('Show Infos'), group: '0' },
            { key: 'showHints', label: nls.localize('theia/markers/showHints', 'Show Hints'), group: '0' },
            { key: 'activeFile', label: nls.localizeByDefault('Show Active File Only'), group: '1' },
            { key: 'useFilesExclude', label: nls.localizeByDefault('Hide Excluded Files'), group: '1' }
        ];

        return opts.map(({ key, label, group }) => ({
            value: key,
            label: creatLabel(label, filters[key as FilterKeys]),
            group,
            onClick: () => clickHandler(key as FilterKeys)
        }));
    }, [filters]);
```